packages/react-ui-core/src/AutoSuggestField/AutoSuggestField.js

```diff
@@ -0,0 +1,205 @@
+import React, { Component, createElement } from 'react'
+import { parseArgs } from '@rentpath/react-ui-utils'
+import debounce from 'lodash/debounce'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import { Dropdown } from '../Dropdown'
+import { Menu } from '../Menu'
+import InputAnchor from './InputAnchor'
+import { Button } from '../Button'
+
+const ENTER = 13
+const ESCAPE = 27
+
+@themed(/^AutoSuggestField/, {
+  pure: true,
+})
+
+export default class AutoSuggestField extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    theme: PropTypes.object,
+    onInput: PropTypes.func,
+    suggestions: PropTypes.node,
+    resetButton: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func,
+      PropTypes.object,
+    ]),
+    onAfterReset: PropTypes.func,
+    submitButton: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func,
+      PropTypes.object,
+    ]),
+    onSubmit: PropTypes.func,
+    onSelection: PropTypes.func,
+    value: PropTypes.string,
+    placeholder: PropTypes.string,
+    suggestionValueSelector: PropTypes.func,
+    onSelectionHover: PropTypes.func,
+    submitOnSelection: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    theme: {},
+    value: '',
+    onAfterReset: () => {},
```

Comment:
Do we really need to default these or can we just check for the existence and execute based on that instead?

Reply:
This has been used in other places in the codebase

Reply:
Well, in some cases I would say it's fine.  Also, just because it's in other places in code doesn't mean those places are correct 😄